backend/user/middleware.go

```diff
@@ -228,7 +234,13 @@ func NewUserMiddleware(queries *sqlc.Queries, remoteCache *remotecache.Client, l
 				roles = append(roles, userRoles...)
 			}
 
-			u.Roles = roles
+			if !u.EmailVerified {
```

Comment:
Any reason not to skip the "GetRolesForEmail" step instead of doing this filtering after?